### PR TITLE
Global shortcuts fixes and tests

### PR DESCRIPTION
**Pull request by jadahl**
While adding basic testing to global shortcuts, I ran into a bunch of issues that were fixed along the way (unnecessary options filtering, refcount issues, etc).

One of the tests will race twice a year, but I hope we can change the test if https://github.com/flatpak/xdg-desktop-portal/issues/1054 can be fixed by changing the meaning of the timestamps in the (de)activated signals.

### CODE REVIEW

**Comment by jadahl** [comment]
Cc: @aleixpol mind reviewing?

File: tests/templates/meson.build [hunk 1]
```diff
@@ -2,6 +2,7 @@ template_files = [
   '__init__.py',
   'email.py',
   'remotedesktop.py',
+  'globalshortcuts.py',
```
Comment (GeorgesStavracas) [inline tests/templates/meson.build:5]: Might be better to sort alphabetically here

**Review by GeorgesStavracas — APPROVED** [review]
Just one minor nit. LGTM otherwise. A bit disappointing that so many `g_variant_ref_sink()`s had to be sprinkled in the file, but so is life.

**Comment by jadahl** [comment]
> A bit disappointing that so many `g_variant_ref_sink()`s had to be sprinkled in the file, but so is life.

An alternative is to do `g_steal_pointer()` in the places that caused the double-frees (e.g. `xdp_dbus_impl_global_shortcuts_call_bind_shortcuts()`)

**Comment by GeorgesStavracas** [comment]
> An alternative is to do g_steal_pointer() in the places that caused the double-frees (e.g. xdp_dbus_impl_global_shortcuts_call_bind_shortcuts())

Now that you mention; I think I'd prefer that :slightly_smiling_face:

**Review by aleixpol — APPROVED** [review]
That is very cool, thank you very much, @jadahl.

**Comment by jadahl** [comment]
> Now that you mention; I think I'd prefer that slightly_smiling_face

Done.

> 23/23 xdg-desktop-portal:pytest / pytest                             OK             57.86s

The timeout seems to happen at 60s - this will start to be flaky I think. Would be nice if the pytest suite could be split up per portal.

**Comment by GeorgesStavracas** [comment]
> The timeout seems to happen at 60s - this will start to be flaky I think. Would be nice if the pytest suite could be split up per portal.

Yes, but I wouldn't block this PR on that :)

**Comment by jadahl** [comment]
> Yes, but I wouldn't block this PR on that :)

Pushed a commit that bumps the timeout to 120s. Will give us some breathing room.

**Pull request merged**